### PR TITLE
feat: setup/teardown localenv with one docker compose

### PR DESCRIPTION
**Pull request by sublimator**
The start/stop scripts are a little slow so I looked what I could do to speed things up, doing some operations in parallel, using shell backgrounding:

```json
"localenv:start:amped": "(pnpm localenv:network:create) && (pnpm localenv:tigerbeetle up -d & pnpm localenv:cloud-nine-wallet up -d & pnpm localenv:happy-life-bank up -d & wait) && (echo rafiki started)",
"localenv:stop:amped": "(pnpm localenv:happy-life-bank down & pnpm localenv:cloud-nine-wallet down & pnpm localenv:tigerbeetle down & wait) && (pnpm localenv:network:remove) && (echo rafiki stopped)",
```

This worked, but the output got all mixed up.  I tried simply starting all the files together  but then realized that docker compose simply merges all the files into one namespace for services, so each file has to have uniquely named services. I then renamed some of the conflicting services in C9/HLB and added a network/docker-compose.yml file. 

I added these package.json scripts:
```json
"localenv:compose": "docker compose -f ./localenv/cloud-nine-wallet/docker-compose.yml -f ./localenv/happy-life-bank/docker-compose.yml ${BEFORE_NETWORK:-}  -f ./localenv/network/docker-compose.yml",
"localenv:compose:tb": "BEFORE_NETWORK='-f ./localenv/tigerbeetle/docker-compose.yml --env-file ./localenv/tigerbeetle/.env.tigerbeetle' pnpm localenv:compose",
```
| Command | Description |
| --- | --- |
| `pnpm localenv:compose config` | Show all merged config (with Tigerbeetle) |
| `pnpm localenv:compose up` | Start (with Tigerbeetle) |
| `pnpm localenv:compose up -d` | Start (with Tigerbeetle) detached |
| `pnpm localenv:compose down` | Down (with Tigerbeetle) |
| `pnpm localenv:compose down --volumes` | Down and kill volumes (with Tigerbeetle) |
| `pnpm localenv:compose config` | Show all merged config |
| `pnpm localenv:compose:psql up` | Start (with Postgresql) |
| `pnpm localenv:compose:psql up -d` | Start (with Postgresql) detached |
| `pnpm localenv:compose:psql down` | Down (with Postgresql) |
| `pnpm localenv:compose:psql down --volumes` | Down (with Postgresql) and kill volumes |

Robo says:

This PR introduces several changes to the project:

1. Improves performance by allowing the setup and take down of services to run in parallel.
1. Direct access to compose command flags 
1. Enables starting up the environment in non-detached mode, making it easier to monitor logs and the status of services.
1. A new merged/docker-compose.yml file is created to serve as the final configuration file for the entire environment. Its purpose is to bring together all individual local environment configurations into a single, unified setup.
1. The services and some volumes have also been renamed by adding a prefix (e.g., cloud-nine-, happy-life-, or shared-) to these elements. This is to give them unique names in the context of the merged setup.

image

### CODE REVIEW

**Comment by sabineschaller** [comment]
And? Does it speed things up?

**Comment by sublimator** [comment]
> And? Does it speed things up?

Yes, definitely! Try it

**Comment by sublimator** [comment]
> Or rather, that there are containers called cloud-nine-wallet-hlb-...

Yeah, I was thinking, you could add "name: rafiki-localenv" to the network/docker-compose.yml 
Kind of hacky, but it's merged last ...

**Comment by sublimator** [comment]
Try again

**Comment by sublimator** [comment]
Robo says:

This PR introduces several changes to the project:

1. In cloud-nine-wallet/docker-compose.yml, happy-life-bank/docker-compose.yml, and tigerbeetle/docker-compose.yml, the version field is replaced with a name field to give unique names to these compose files (short codes for when loading the files individually). The services and some volumes have also been renamed by adding a prefix (e.g., cloud-nine-, happy-life-, or shared-) to these elements. This is to give them unique names in the context of the merged setup.
1. In happy-life-bank/docker-compose.yml, docker compose setup is updated with unique service names formed by adding a happy-life- prefix.
1. A new merged/docker-compose.yml file is created to serve as the final configuration file for the entire environment. Its purpose is to bring together all individual local environment configurations into a single, unified setup.
1. In package.json, two new scripts are added: localenv:compose and localenv:compose:tb. These scripts are used to run the Docker Compose command with the merged configuration, allowing you to start and manage containers for all services in the project at once. The localenv:compose script runs docker compose for the cloud nine wallet and happy life bank compose files, and the `localenv:compose:tb' script runs it with the tigerbeetle compose file as well.
These changes make it easier to manage and configure services in the project, and allow you to run all services in a single Docker environment.

image

File: package.json [hunk 1]
```diff
@@ -18,19 +18,8 @@
     "check:prettier": "prettier --check .",
     "clean": "find . -name 'node_modules' -type d -prune -exec rm -rf '{}' +",
     "build": "tsc --build",
-    "localenv:network:create": "docker network create --driver=bridge --subnet=10.5.0.0/24 --gateway=10.5.0.1 rafiki",
-    "localenv:network:remove": "docker network remove rafiki",
-    "localenv:cloud-nine-wallet": "docker compose -f ./localenv/cloud-nine-wallet/docker-compose.yml",
-    "localenv:happy-life-bank": "docker compose  -f ./localenv/happy-life-bank/docker-compose.yml",
-    "localenv:tigerbeetle": "docker compose -f ./localenv/tigerbeetle/docker-compose.yml --env-file ./localenv/tigerbeetle/.env.tigerbeetle",
-    "localenv:build": "pnpm localenv:tigerbeetle build && pnpm localenv:cloud-nine-wallet build && pnpm localenv:happy-life-bank build",
```
Comment (sabineschaller) [inline package.json:26]: Can we please keep that command, obviously changing it to use the new structure

**Review by sabineschaller — COMMENTED** [review]
I wasn't able to run it. I got
```
$ pnpm localenv:compose                                                       

> @interledger/rafiki@1.0.0-alpha.1 localenv:compose /Users/sabine/git/coil/rafiki
> BEFORE_MERGED='-f ./localenv/tigerbeetle/docker-compose.yml' pnpm localenv:compose:psql


> @interledger/rafiki@1.0.0-alpha.1 localenv:compose:psql /Users/sabine/git/coil/rafiki
> docker compose -f ./localenv/cloud-nine-wallet/docker-compose.yml -f ./localenv/happy-life-bank/docker-compose.yml ${BEFORE_MERGED:-}  -f localenv/merged/docker-compose.yml


Usage:  docker compose [OPTIONS] COMMAND

Docker Compose
...
```

File: package.json (hunk 1 above)
Comment (sublimator) [reply package.json:26]: you just use `localenv:compose build`

File: package.json [hunk 2]
```diff
@@ -18,19 +18,8 @@
...
-    "localenv:start": "pnpm localenv:network:create && pnpm localenv:tigerbeetle up -d && pnpm localenv:cloud-nine-wallet up -d && pnpm localenv:happy-life-bank up -d",
-    "localenv:stop": "pnpm localenv:happy-life-bank down && pnpm localenv:cloud-nine-wallet down && pnpm localenv:tigerbeetle down && pnpm localenv:network:remove",
-    "localenv:dbvolumes:remove": "docker volume rm cloud-nine-wallet_database-data && docker volume rm tigerbeetle_tigerbeetle-data",
-    "localenv:psql:build": "pnpm localenv:cloud-nine-wallet build && pnpm localenv:happy-life-bank build",
-    "localenv:psql:start": "pnpm localenv:network:create && pnpm localenv:cloud-nine-wallet up -d && pnpm localenv:happy-life-bank up -d",
-    "localenv:psql:stop": "pnpm localenv:cloud-nine-wallet down && pnpm localenv:happy-life-bank down && pnpm localenv:network:remove",
-    "localenv:psql:dbvolumes:remove": "docker volume rm cloud-nine-wallet_database-data",
+    "localenv:compose:psql": "docker compose -f ./localenv/cloud-nine-wallet/docker-compose.yml -f ./localenv/happy-life-bank/docker-compose.yml ${BEFORE_MERGED:-}  -f localenv/merged/docker-compose.yml",
```
Comment (mkurapov) [inline package.json:21]:
```suggestion
    "localenv:compose:psql": "docker compose -f ./localenv/cloud-nine-wallet/docker-compose.yml -f ./localenv/happy-life-bank/docker-compose.yml ${BEFORE_MERGED:-} -f ./localenv/merged/docker-compose.yml",
```
just for consistency

**Review by mkurapov — COMMENTED** [review]
looks good :+1:

**Comment by sublimator** [comment]
On mobile

Please free feel to take over PR to merge

I merged the suggestions before which apparently prettier is complaining
about?

**Review by mkurapov — APPROVED** [review]

**Comment by sublimator** [comment]
Wooo, thanks @mkurapov

**Pull request merged**